### PR TITLE
Fix Bug 1307217, add link to section elements to improve ux

### PR DESCRIPTION
**Pull request by schalkneethling**
@hobinjk @jwhitlock r?

Example of what this looks like on a page:

screen shot 2018-07-30 at 15 11 26

### CODE REVIEW

File: kuma/static/js/components/local-anchor.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+/**
+ * Find all level two and three headings, and append a link
+ * element that links directly to the heading
+ */
+function appendLocalAnchors() {
```
Comment (schalkneethling) [inline kuma/static/js/components/local-anchor.js]: I have no idea why this reads `unction` :) My local file definitely does not.

Comment (schalkneethling) [reply kuma/static/js/components/local-anchor.js]: Oh, and if I click on the view button, the resulting view does show `function`. Guess it's a Github issue.

Comment (jwhitlock) [reply kuma/static/js/components/local-anchor.js]: It looks like ``function`` to me. Update or uninstall those plugins!

Does ``appendLocalAnchors`` mean HTML element order? The link appears to prepend the headings for me.

**Comment by schalkneethling** [comment]
> Visually, I was expecting something like the Kuma docs, where the anchor indicator appears when hovering over the title.

Agreed. It would be good to get some user feedback on it. My initial thought was to have that be the behaviour but:

1. The entire heading is not wrapped in the anchor so, I would have to do the hover in a hack like way by changing the pointer when somebody hovers over the heading. It would not have the "natural" behaviour as it is not actually a link.
2. As this is new, perhaps people will notice it more readily, and thus get the benefit quicker
3. This way it also works on mobile

We can of course change the behaviour if we get a lot of negative feedback.

File: kuma/static/js/components/local-anchor.js (hunk 1 above)
Comment (schalkneethling) [reply kuma/static/js/components/local-anchor.js]: Good call, `prependLocalAnchors` makes more sense. I was initially appending, and did not think to change the name when I switched to prepending ;)

**Comment by schalkneethling** [comment]
@jwhitlock update. Thanks @jwhitlock and @hobinjk for the reviews!

**Comment by jezdez** [comment]
ZOMG!!1! Thank you for adding those! 😻

**Pull request merged**